### PR TITLE
Project restructure to be more in line with golang standards

### PR DESCRIPTION
**Pull request by tylarb**
- Need to improve build scripts
- Need to improve travis CI

Authored by Tyler Ramer <tramer@pivotal.io>

### CODE REVIEW

**Review by faisaltheparttimecoder — COMMENTED** [review]
Suggestion: 

I believe there should not be a duplication of directory naming with the go directory standard , like folder named "pkg" ..

File: scripts/build.sh
Comment (faisaltheparttimecoder) [inline scripts/build.sh:8]: Its good to have the OS information ( i.e GOOS & GOARCH ) with build, since a single build might not work with all the OS. so specifying the OS that we are sure we support might help..

Have a look at this blog for all the GOOS & GOARCH

https://www.digitalocean.com/community/tutorials/how-to-build-go-executables-for-multiple-platforms-on-ubuntu-16-04

Basically what I mean is , the the above build will build the execute package based on the OS architecture where we ran the build.. for eg.s if you run on the mac terminal then it will build a execute binary that works only on mac and its architecture.

File: pkg/db/db.go
Comment (faisaltheparttimecoder) [inline pkg/db/db.go:23]: *sql.DB is thread-safe and therefore can be accessed concurrently, if you define it within a function you are basically starting a pointer and then passing a new pointer , which doesn't make a lot of sense.

So having this defined globally can help in every sub function to know the connection pointer and doesn't chew up memory ..

File: cmd/gpmt/root.go
Comment (faisaltheparttimecoder) [inline cmd/gpmt/root.go:41]: Did you mean wrapper in the sense ( db.go ) or the ConnString type struct ?

File: pkg/db/db.go
Comment (faisaltheparttimecoder) [inline pkg/db/db.go]: Its just a placeholder to keep the data and its type, since we don't know what would be the type of the column that is going to send to us when the user sends us a query.

File: pkg/db/db.go
Comment (faisaltheparttimecoder) [inline pkg/db/db.go:29]: Its just a placeholder to keep the data and its type, since we don't know what would be the type of the column that is going to send to us when the user sends us a query.

File: scripts/build.sh
Comment (tylarb) [reply scripts/build.sh:8]: True - the scripts should be developed more, along with testing scripts. This is just an initial project framework.

File: cmd/gpmt/root.go
Comment (tylarb) [reply cmd/gpmt/root.go:41]: I mean in the sense of a db.go file. Maybe we would move it to the internal folder, if we need some easy way of executing commands. But the database/sql golang package is already fairly obfuscated. We will consider further in time.

**Comment by tylarb** [comment]
I used https://github.com/golang-standards/project-layout as the major guide for the directory structure, along with projects such as Moby, Kubernetes, etc. 

Everything in the pkg folder should be
1. Able to be exported (i.e. standalone) 
2. In its own package/folder

**Comment by faisaltheparttimecoder** [comment]
Did knew that they have a project structure now .. cool thanks

**Comment by ielizaga** [comment]
I am all in favour of using https://github.com/golang-standards/project-layout.

The project structure defined in that repository makes sense and it also keeps us from having to document a new one from scratch. 

In my opinion, this is big enough to require all three approvals before merging.

**Review by ielizaga — APPROVED** [review]
I am all in favour of using https://github.com/golang-standards/project-layout.

The project structure defined in that repository makes sense and it also keeps us from having to document a new one from scratch.

In my opinion, this is big enough to require all three approvals before merging.

**Comment by ielizaga** [comment]
Before we merge this, also, we should maybe look in detail at the issues open for that repository, this one in particular is similar to what Faisal was saying before:

https://github.com/golang-standards/project-layout/issues/10

**Comment by tylarb** [comment]
I can understand the logic for moving items up from the pkg folder - although it's not a strict issue. I do think Greenplum's new golang tooling also uses a /[exported lib]  (i.e. not in pkg folder) so I'll restructure around that. It seems a good fit for a small project such as this. Will restructure based on this change and squash+merge later today. 

I think one argument FOR the pkg is like this:
/internal is unexported libraries - not intended for use in other (people's) projects
/vendor is 3rd party libraries
/pkg is exported and can be used in other projects.

**Comment by faisaltheparttimecoder** [comment]
i guess, I'm okay with the current structure we can change the name if the golang standard change their project layout, for now lets merge this..

**Pull request merged**
